sdk/cosmos/azure-cosmos/azure/cosmos/aio/container.py

```diff
@@ -198,16 +199,20 @@ async def create_item(
 
     @distributed_trace_async
     async def read_item(
-        self,
-        item,  # type: Union[str, Dict[str, Any]]
-        partition_key,  # type: Any
-        **kwargs  # type: Any
+            self,
+            item,  # type: Union[str, Dict[str, Any]]
+            partition_key,  # type: Any
+            max_integrated_cache_staleness_in_ms=None,  # type: Optional[int],
+            **kwargs  # type: Any
     ):
         # type: (...) -> Dict[str, Any]
         """Get the item identified by `item`.
 
         :param item: The ID (name) or dict representing item to retrieve.
         :param partition_key: Partition key for the item to retrieve.
+        **Provisional** parameter max_integrated_cache_staleness_in_ms
```

Comment:
@annatisch  - with this `Provisional` API mention, would we be fine GAing our SDK even if this stays `Provisional`?

Reply:
There are a couple of questions still open around that.
For example - the SDK can not go GA pointing to a preview API version.
Given that this parameter effectively amounts to just a header - we can actually unblock customers who want to use this without adding it as an API feature. That way we have time to chose how to surface it in the SDK as a GA feature when the service goes GA, and customers who were already using it remain unbroken.
In this case it would be something like:
```python
client = CosmosClient(..., api_version="preview-version")
client.read_item(..., headers={"DedicatedGatewayCacheStaleness":"10"}
```

Reply:
This is a preview feature but does not require an API version change - it instead requires users to set up the dedicated gateway in their database accounts manually in order to point their service at a different endpoint, so these concerns are taken care of.